### ISSUE TITLE
Auto-correct compressed hash syntax incorrect

### ISSUE THREAD

**Issue by swordfish444** [bbatsov/rubocop#5019]
An issue with compressed hash syntax being incorrectly auto-corrected.

--------

## Expected behavior

`model.save(validate:true)`

## Actual behavior

`model.savevalidate: true`

## Steps to reproduce the problem

```
user = User.new
user.save:validate=>true
```

then run `rubocop --auto-correct`

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by pocke** [comment]
It is a problem of `Style/HashSyntax`

Minimum reproduce
==

`test.rb`

```ruby
a:b => c
```




```bash
$ rubocop --only Style/HashSyntax -a
$ rubocop --only Style/HashSyntax -a
Inspecting 1 file
E

Offenses:

test.rb:1:2: C: [Corrected] Use the new Ruby 1.9 hash syntax.
a:b => c
 ^^^^^
test.rb:1:3: E: unexpected token tCOLON
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
ab: c
  ^

1 file inspected, 2 offenses detected, 1 offense corrected

$ cat test.rb
ab: c
```

**Comment by donjar** [comment]
I cannot reproduce.

```
$ rubocop -a test.rb
Inspecting 1 file
C

Offenses:

test.rb:1:1: C: [Corrected] Missing magic comment # frozen_string_literal: true.
a:b => c
^
test.rb:1:2: C: [Corrected] Put one space between the method name and the first argument.
a:b => c
 
test.rb:1:2: C: [Corrected] Use the new Ruby 1.9 hash syntax.
a:b => c
 ^^^^^
test.rb:2:1: C: [Corrected] Add an empty line after magic comments.
a b: c
^

1 file inspected, 4 offenses detected, 4 offenses corrected

$ cat test.rb
# frozen_string_literal: true

a b: c
```

**Comment by pocke** [comment]
@donjar Can you try `--only` option? 

> test.rb:1:2: C: [Corrected] Put one space between the method name and the first argument.

This correction conceals the bug.

**Comment by donjar** [comment]
Oops, didn't notice that. Yeah now it errors.

```
$ rubocop -a --only Style/HashSyntax test.rb
Inspecting 1 file
E

Offenses:

test.rb:1:2: C: [Corrected] Use the new Ruby 1.9 hash syntax.
a:b => c
 ^^^^^
test.rb:1:3: E: unexpected token tCOLON
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
ab: c
  ^

1 file inspected, 2 offenses detected, 1 offense corrected

$ cat test.rb
ab: c
```

I'll try a fix. :)
